google/cloud/spanner_v1/snapshot.py

```diff
@@ -680,10 +680,13 @@ def partition_read(
         )
 
         trace_attributes = {"table_id": table, "columns": columns}
+        can_include_index = (index != "") and (index is not None)
+        if can_include_index:
+            trace_attributes["index"] = index
+
         with trace_call(
             f"CloudSpanner.{type(self).__name__}.partition_read",
-            self._session,
```

Comment:
Lets keep this session here since we are updating last use time of session on the trace_call method..

Reply:
Done, thanks!